False positive for N805 when using positional-only args

```console
> cat test.py
class Hello:
    def __init__(self, blah, /, something: str, another: str, yeah: str):
        pass

> ruff --select N test.py
test.py:2:33: N805 First argument of a method should be named `self`
Found 1 error.

> ruff --version
ruff 0.0.243
```

--- a/crates/ruff/resources/test/fixtures/pep8_naming/N805.py
+++ b/crates/ruff/resources/test/fixtures/pep8_naming/N805.py
@@ -51,3 +51,11 @@ class MetaClass(ABCMeta):
 
 def func(x):
     return x
+
+
+class PosOnlyClass:
+    def good_method_pos_only(self, blah, /, something: str):
+        pass
+
+    def bad_method_pos_only(this, blah, /, self, something: str):
+        pass

--- a/crates/ruff/src/rules/pep8_naming/rules.rs
+++ b/crates/ruff/src/rules/pep8_naming/rules.rs
@@ -316,7 +316,7 @@ pub fn invalid_first_argument_name_for_method(
     ) {
         return None;
     }
-    let arg = args.args.first()?;
+    let arg = args.posonlyargs.first().or_else(|| args.args.first())?;
     if arg.node.arg == "self" {
         return None;
     }

